power-sequencer/mihawk-cpld.cpp

```diff
@@ -0,0 +1,478 @@
+        }
+        else if (errorcode == 21)
+        {
+            report<PowerOnErrorVdn0PGOODFail>();
+            std::cerr << "Power_on error : Vdn0_pgood fail! \n";
+        }
+        else if (errorcode == 22)
+        {
+            report<PowerOnErrorVdn1PGOODFail>();
+            std::cerr << "Power_on error : Vdn1_pgood fail! \n";
+        }
+        else if (errorcode == 23)
+        {
+            report<PowerOnErrorP1V5PGOODFail>();
+            std::cerr << "Power_on error : P1V5_pgood fail! \n";
+        }
+        else if (errorcode == 24)
+        {
+            report<PowerOnErrorVio0PGOODFail>();
+            std::cerr << "Power_on error : Vio0_pgood fail! \n";
+        }
+        else if (errorcode == 25)
+        {
+            report<PowerOnErrorVio1PGOODFail>();
+            std::cerr << "Power_on error : Vio1_pgood fail! \n";
+        }
+        else if (errorcode == 26)
+        {
+            report<PowerOnErrorVdd0PGOODFail>();
+            std::cerr << "Power_on error : Vdd0_pgood fail! \n";
+        }
+        else if (errorcode == 27)
+        {
+            report<PowerOnErrorVcs0PGOODFail>();
+            std::cerr << "Power_on error : Vcs0_pgood fail! \n";
+        }
+        else if (errorcode == 28)
+        {
+            report<PowerOnErrorVdd1PGOODFail>();
+            std::cerr << "Power_on error : Vdd1_pgood fail! \n";
+        }
+        else if (errorcode == 29)
+        {
+            report<PowerOnErrorVcs1PGOODFail>();
+            std::cerr << "Power_on error : Vcs1_pgood fail! \n";
+        }
+        else if (errorcode == 30)
+        {
+            report<PowerOnErrorVddr0PGOODFail>();
+            std::cerr << "Power_on error : Vddr0_pgood fail! \n";
+        }
+        else if (errorcode == 31)
+        {
+            report<PowerOnErrorVtt0PGOODFail>();
+            std::cerr << "Power_on error : Vtt0_pgood fail! \n";
+        }
+        else if (errorcode == 32)
+        {
+            report<PowerOnErrorVddr1PGOODFail>();
+            std::cerr << "Power_on error : Vddr1_pgood fail! \n";
+        }
+        else if (errorcode == 33)
+        {
+            report<PowerOnErrorVtt1PGOODFail>();
+            std::cerr << "Power_on error : Vtt1_pgood fail! \n";
+        }
+        else if (errorcode == 34)
+        {
+            report<PowerOnErrorGPU0PGOODFail>();
+            std::cerr << "Power_on error : GPU0_pgood fail! \n";
+        }
+        else if (errorcode == 35)
+        {
+            report<PowerOnErrorGPU1PGOODFail>();
+            std::cerr << "Power_on error : GPU1_pgood fail! \n";
+        }
+        else if (errorcode == 170)
+        {
+            report<PowerOnErrorPSU0PSU1PGOODFail>();
+            std::cerr << "Power_on error : Psu0&PSU1_pgood fail! \n";
+        }
+    }
+}
+
+void MihawkCPLD::analyze()
+{
+    bool psuError = checkPSUdevice();
+    if (psuError)
+    {
+        // If PSU not present, report the error log & event.
+        report<PSUPresentError>();
```

Comment:
there is no need to indicate which PSU wasn't present?

Reply:
This part of the judgment is based on the fact that both PSUs have fails(psu0 & psu1 aren't present) at the same time, resulting in a failure to power on